backend-api/src/functions/asyncTxmaEvent/asyncTxmaEventHandler.ts

```diff
@@ -32,6 +50,34 @@ export async function lambdaHandlerConstructor(
     });
     return badRequestResponse("invalid_request", errorMessage);
   }
+  const { sessionId } = validateRequestBodyResult.value;
+  const eventService = dependencies.getEventService(config.TXMA_SQS);
+
+  const sessionRegistry = dependencies.getSessionRegistry(
+    config.SESSION_TABLE_NAME,
+  );
+  const getSessionResult = await sessionRegistry.getSession(
+    sessionId,
+    new TxMAEvent(),
+  );
+  const { ipAddress, txmaAuditEncoded } = getAuditData(event);
+  const eventData = {
+    sessionId,
+    issuer: config.ISSUER,
+    ipAddress,
+    txmaAuditEncoded,
+  };
+  if (getSessionResult.isError) {
+    return handleGetSessionError(
+      eventService,
+      getSessionResult.value,
+      eventData,
+    );
+  }
+  const sessionAttributes = getSessionResult.value;
+
+  const isSessionValid = validateSession(sessionAttributes);
+  if (!isSessionValid) return handleSessionNotFound(eventService, eventData);
```

Comment:
```suggestion
  if (!validateSession(sessionAttributes)) return handleSessionNotFound(eventService, eventData);
```

Entirely non-blocking. My thought is we don't use the const anywhere else and the function name provides enough meaning imo, happy for you to resolve tho

Reply:
This or the validateSession could also return a Result and then we follow the typical isError pattern